sparse_strips/vello_common/src/glyph.rs

```diff
@@ -0,0 +1,235 @@
+    pub local_transform: Affine,
+}
+
+/// Trait for types that can render glyphs.
+pub trait GlyphRenderer {
+    /// Fill glyphs with the current paint and fill rule.
+    fn fill_glyph(&mut self, glyph: PreparedGlyph<'_>);
+
+    /// Stroke glyphs with the current paint and stroke settings.
+    fn stroke_glyph(&mut self, glyph: PreparedGlyph<'_>);
+}
+
+/// A builder for configuring and drawing glyphs.
+#[derive(Debug)]
+pub struct GlyphRunBuilder<'a, T: GlyphRenderer + 'a> {
+    run: GlyphRun<'a>,
+    renderer: &'a mut T,
+}
+
+impl<'a, T: GlyphRenderer + 'a> GlyphRunBuilder<'a, T> {
+    /// Creates a new builder for drawing glyphs.
+    pub fn new(font: Font, transform: Affine, renderer: &'a mut T) -> Self {
+        Self {
+            run: GlyphRun {
+                font,
+                font_size: 16.0,
+                transform,
+                glyph_transform: None,
+                hint: true,
+                normalized_coords: &[],
+            },
+            renderer,
+        }
+    }
+
+    /// Set the font size in pixels per em.
+    pub fn font_size(mut self, size: f32) -> Self {
+        self.run.font_size = size;
+        self
+    }
+
+    /// Set the per-glyph transform. Can be used to apply skew to simulate italic text.
+    pub fn glyph_transform(mut self, transform: Affine) -> Self {
+        self.run.glyph_transform = Some(transform);
+        self
+    }
+
+    /// Set whether font hinting is enabled.
+    pub fn hint(mut self, hint: bool) -> Self {
+        self.run.hint = hint;
+        self
+    }
+
+    /// Set normalized variation coordinates for variable fonts.
+    pub fn normalized_coords(mut self, coords: &'a [NormalizedCoord]) -> Self {
+        self.run.normalized_coords = bytemuck::cast_slice(coords);
+        self
+    }
+
+    /// Consumes the builder and fills the glyphs with the current configuration.
+    pub fn fill_glyphs(self, glyphs: impl Iterator<Item = Glyph>) {
+        self.render(glyphs, Style::Fill);
+    }
+
+    /// Consumes the builder and strokes the glyphs with the current configuration.
+    pub fn stroke_glyphs(self, glyphs: impl Iterator<Item = Glyph>) {
+        self.render(glyphs, Style::Stroke);
+    }
+
+    fn render(self, glyphs: impl Iterator<Item = Glyph>, style: Style) {
+        let font =
+            skrifa::FontRef::from_index(self.run.font.data.as_ref(), self.run.font.index).unwrap();
+        let outlines = font.outline_glyphs();
+        let size = Size::new(self.run.font_size);
+        let hinting_instance = if self.run.hint {
+            // Only transformations including uniform scaling and translation can be hinted.
+            // Rotated, skewed, and other transformations cannot be hinted.
+            let [a, b, c, d, _, _] = self.run.transform.as_coeffs();
+            if a == d && b == 0.0 && c == 0.0 {
+                // TODO: Cache hinting instance.
+                HintingInstance::new(&outlines, size, self.run.normalized_coords, HINTING_OPTIONS)
```

Comment:
So just to really lay this out, currently we scale glyphs post-hinting. Which as an MVP is *fine*, but it is definitely wrong behaviour (because you're not really hinting unless you have a scale of exactly 1.0!).

We should have a comment here, probably.

Reply:
Good callout. Added comment in https://github.com/linebender/vello/pull/883/commits/8284434061acb5f6f89c26c9daa014c88d3169be. I also only applied hinting when scale is 1.0. Immediately after this, I'll prepare https://github.com/taj-p/vello/pull/5 for review.